libres/tests/analysis/test_update.cpp

```diff
@@ -200,6 +207,193 @@ SCENARIO("Running analysis update without row scaling on linear model",
                 REQUIRE(d_posterior_ml[n_sd - 1] < eps);
             }
         }
+
+        WHEN("Row scaling factor is 0 for both parameters") {
+            row_scaling_type *row_scaling = row_scaling_alloc();
```

Comment:
For these `xxx_type * ptr = xxx_alloc()` calls you can choose to use `auto * ptr = xxx_alloc(); - I guess that would be more idiomatic C++.

Reply:
Thanks, I agree.